src/gui/MainWindow.cpp

```diff
@@ -522,8 +522,10 @@ void MainWindow::printPreviewClicked()
 
 void MainWindow::showOpenImageDialog()
 {
-    auto path = QFileDialog::getOpenFileName(this, tr("Open Image"), mSavePathProvider.saveDirectory(), tr("Image Files (*.png *.jpg *.bmp)"));
-	loadImageFromFile(path);
+	auto pathList = QFileDialog::getOpenFileNames(this, tr("Open Images"), mSavePathProvider.saveDirectory(), tr("Image Files (*.png *.jpg *.bmp)"));
```

Comment:
Can you move this `QFileDialog::getOpenFileName` to the `FileDialogAdapter` and then access it via the `IFileDialogAdapter` interface? Maybe we should rename the `IFileDialogAdapter` to `IFileDialogService`